### ISSUE TITLE
Make sure n:m relationships only exist once (for selected tables)

### ISSUE THREAD

**Issue by tmfrnz** [dumparkltd/advocacy-tracker-server#33]
We want to make sure that the db does not contain duplicate relationships (unless they have distinguishing attributes)

Specifically this affects the following tables

### memberships

- [ ] only one combination of `member_id` and `other_member_id` allowed (a can be member of b only once), including the inverse relationship (if a is member of b then b can't also be member of a)

### actor_categories

- [ ] only one combination of `actor_id` and `category_id` allowed

### measure_categories

- [ ] only one combination of `measure_id` and `category_id` allowed

### measure_indicators

- [ ] only one combination of `measure_id` and `indicator_id` allowed

### measure_measures

- [ ] only one combination of `measure_id` and `other_measure_id` allowed (interpreted as a is child of b (other), including the inverse relationship (if a is child of b, b can't be also child of a)

### measure_resources

- [ ] only one combination of `measure_id` and `resource_id` allowed

### user_actors

- [ ] only one combination of `user_id` and `actor_id` allowed

### user_measures

- [ ] only one combination of `user_id` and `measure_id` allowed

**Comment by tmfrnz** [comment]
(note that multiple relationships between the same 2 entities are allowed in 'measure_actors' and 'actor_measures' as there could be different values or dates

**Comment by parndt** [comment]
I created a PR for this - some of them, it seems, were already covered.

**Comment by parndt** [comment]
Thanks @tmfrnz
